packages/graph-retriever/src/graph_retriever/adapters/in_memory.py

```diff
@@ -113,7 +113,11 @@ def _matches(self, filter: dict[str, Any] | None, content: Content) -> bool:
             return True
 
         for key, filter_value in filter.items():
-            content_value = content.metadata.get(key, SENTINEL)
+            content_value = content.metadata
+            for key_part in key.split("."):
```

Comment:
Split keys on `.`. This is questionable -- some people might put `a.b` in the key.